### ISSUE TITLE
[REVIEW]: survPen: an R package for hazard and excess hazard modelling with multidimensional penalized splines

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1434]
**Submitting author:** @fauvernierma (<a href="http://orcid.org/0000-0002-3809-8248">Mathieu Fauvernier</a>)
**Repository:** <a href="https://github.com/fauvernierma/survPen" target ="_blank">https://github.com/fauvernierma/survPen</a>
**Version:** v1.2.0
**Editor:** @csoneson
**Reviewer:** @corybrunson, @seabbs
**Archive:** <a href="https://doi.org/10.5281/zenodo.3372656" target="_blank">10.5281/zenodo.3372656</a>

## Status

[![status](http://joss.theoj.org/papers/ddf9a5092cd03aa05528cef3740c59a0/status.svg)](http://joss.theoj.org/papers/ddf9a5092cd03aa05528cef3740c59a0)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/ddf9a5092cd03aa05528cef3740c59a0"><img src="http://joss.theoj.org/papers/ddf9a5092cd03aa05528cef3740c59a0/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/ddf9a5092cd03aa05528cef3740c59a0/status.svg)](http://joss.theoj.org/papers/ddf9a5092cd03aa05528cef3740c59a0)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/fauvernierma/survPen" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@corybrunson & @seabbs, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @csoneson know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/fauvernierma/survPen">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v1.2.0
- [x] **Authorship:** Has the submitting author (@fauvernierma) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/fauvernierma/survPen">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v1.2.0
- [x] **Authorship:** Has the submitting author (@fauvernierma) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @corybrunson, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01434/joss.01434/10.21105.joss.01434.pdf)

**Comment by csoneson** [comment]
@corybrunson, @seabbs - please perform your review in this issue, by updating your checklist above and posting comments (also see the additional instructions in the posts above). If you have any questions, don't hesitate to ping me.

**Comment by corybrunson** [comment]
@csoneson i apologize for the delay on my part. I won't be able to review the submission tomorrow but i expect to this week.

**Comment by seabbs** [comment]
Hi @fauvernierma,

First off this package looks great and I particularly like the detailed vignette. 👍 

Broken down comments by section. 

## General Checks

- [ ]  License needs references to your actual package in the appropriate places. fauvernierma/survPen#2
- [ ] Whilst your released version (i.e CRAN) matches your dev version (GitHub) I would ideally see some kind of explicit archiving of given releases. See fauvernierma/survPen#3 for details - this is not an acceptance blocker. 
- [ ] Authorship - I am only seeing commits from the first author.  Is a contribution statement in the paper a good idea @csoneson. See fauvernierma/survPen#4

## Software paper

- [ ] Statement of need - needs some fine tuning. See  fauvernierma/survPen#5
- [ ] References - missing DOI. See fauvernierma/survPen#6

## Documentation

- [ ] Statement of need - as for fauvernierma/survPen#5.  This needs to be in the package description and the readme. Vignette is very clear but lacks a quick intro as to why you want to use this package.
- [ ] Documentation - see  fauvernierma/survPen#9
- [ ] Tests - need automated unit tests.  See fauvernierma/survPen#8
- [ ] Community guidelines - missing. See fauvernierma/survPen#7

## Functionality

- [ ] Performance - paper says `survPen` is faster + more stable than alternatives. An example of this is needed. See fauvernierma/survPen#10
- [ ] Functionality - on the surface this all looks great. Unfortunately, I am not qualified to assess the underlying methods. As above I would like to see a comparison to alternative software solutions (even solutions not using splines) to see the improvement this method gives.

## Misc

* The code base looks robust and well implemented. 
* `survPen` has minimal dependencies which makes installation etc. a breeze. 
- [ ] [*optional*] You state in the description that at least R `3.5.0` is required. Is this true? It might be good to allow for some earlier R versions if possible
- [ ] [*optional*] I prefer each function (or group of functions) to have their own R script.

## Summary

Overall `survPen` looks like a great package and a useful contribution to the R universe 👍. I am looking forward to conducting future analyses using it. However, there are several outstanding issues with testing, documentation and performance/functionality (outlined above) that need resolving before I can recommend this package. Happy to discuss these points further in the relevant issues and provide additional support as required :smile:

**Comment by fauvernierma** [comment]
Thank you very much for @seabbs for the review

@csoneson, should I go to "pull request" in github to address each issue or can I just edit the file ?
And also am I authorized to close the issues myself when I consider that they have been addressed ?

Thanks

**Comment by csoneson** [comment]
Thanks for your review @seabbs! Regarding the author contribution statement, we currently don't require that. Ultimately, authors are responsible for deciding which contributions are sufficient for co-authorship (see more at https://joss.readthedocs.io/en/latest/review_criteria.html#authorship). @fauvernierma, since it's difficult to judge from the commit history in your repository - could you please confirm that all the authors qualify for inclusion with respect to these criteria, to address @seabbs's comment?

**Comment by csoneson** [comment]
@fauvernierma: You don't have to make a separate pull request for each fix. It may be useful to keep each fix in a separate commit, and refer to the commit in the corresponding issue. This would keep things modular and make it easy for the reviewers to see what you changed. You can also close the issues in your repository - the evaluation from the reviewers will be recorded in the checklist in the beginning of this issue; this is where they will record that a fix is satisfactory to address their comments.

**Comment by fauvernierma** [comment]
Ok, thanks @csoneson .

@seabbs, as for the authorship, the four other authors/contributors and I are part of a survival working group within our research department. Two of them (Laurent Remontet and Nadine Bossard) are indeed my thesis advisors.
Laurent Roche and Laurent Remontet are cited as authors as they intervened in the core development of the package (especially in linear algebra and matrix derivatives).
As for Zoé Uhry and Nadine Bossard, they contributed to the writing of the documentation and gave important pieces of advice relative to the package functionalities.

I hope this addresses your comment @seabbs

**Comment by seabbs** [comment]
Yes, that is great #fauvernierma - thanks :smile:

**Comment by corybrunson** [comment]
@fauvernierma this might not be appropriate to raise as an issue on the code repo: Could you explain the value `GPL (>= 2)` in the `DESCRIPTION` file? The `LICENSE` file describes GPL-3, which is acceptable to CRAN (see [here](https://www.r-project.org/Licenses/)), but i understand GPL-3 to be slightly more restrictive than GPL-2 and i haven't seen this notation before.

**Comment by fauvernierma** [comment]
Hello @corybrunson 
As the new LICENSE file for survPen now uses GPL-3 (thanks to point raised by @seabbs), this old notation is going to change in the DESCRIPTION file.

**Comment by seabbs** [comment]
@csoneson - there is a methods paper under review for this package that is not available to the reviewers/users. From the references to it in the documentation (and from what @fauvernierma  has said) it contains much of the detail confirming the package functionality/performance.

What is the best approach to avoid replicating work for  @fauvernierma but still providing reviewers/users with assurances that everything works as advertised?

**Comment by corybrunson** [comment]
Regarding @seabbs's miscellaneous comment above, i was able to install **survPen** and reproduce the vignette using R version 3.3.2. I agree that reducing the version dependency is optional, and i wouldn't necessarily recommend it without continuous integration support or unless @fauvernierma can test it using an earlier version on their own machine.

**Comment by csoneson** [comment]
@seabbs - thanks for raising the question about the associated methods paper. The software functionality should be possible to review with publicly available material; thus, either the methods paper should be deposited on a preprint server, or enough material to review functionality and performance claims should be included in the documentation of the software. 
@fauvernierma - please note that all references that are cited in the paper need to be publicly available (with a DOI or similar identifier).

**Comment by csoneson** [comment]
👋 @fauvernierma - just wondered if you could give us a status update on how the revision of your submission is going? Of course, please don't hesitate to reach out if you have questions!

**Comment by fauvernierma** [comment]
Hello @csoneson

The revision has been going quite well so far although I have to admit I did not anticipate the issue with the method paper. Speaking of which, the status of my revised manuscript now reads "With Editor" so it is just a matter of hours until I get the answer.
Hopefully the method paper will be accepted but it is also likely that I will have some other details to fix.

Meanwhile I am finishing my thesis manuscript and the due date is on the 25th of this month and it is going to be pretty complicated for me to work on the JOSS review. 

Is it ok if we put it on hold until the 25th ?

Thank you

**Comment by csoneson** [comment]
Thanks for the update @fauvernierma. I think that's ok - I will ask whedon to send an automated reminder in 3 weeks, after your thesis deadline.

**Comment by csoneson** [comment]
@whedon remind @author in three weeks

**Comment by whedon** [comment]
@author doesn't seem to be a reviewer or author for this submission.

**Comment by csoneson** [comment]
@whedon remind @fauvernierma in three weeks

**Comment by whedon** [comment]
Reminder set for @fauvernierma in three weeks

**Comment by whedon** [comment]
:wave: @fauvernierma, please update us on how things are progressing here.

**Comment by fauvernierma** [comment]
Hello everyone,

My method article has finally been accepted for publication in the Journal of the Royal Statistical Society Series C. I will soon be able to provide a DOI.

Concerning the ongoing issues I will respond as soon as possible (probably this Friday or on Monday)

Have a nice day

**Comment by csoneson** [comment]
👋 @fauvernierma, could you give us an update on how the revision is progressing?

**Comment by fauvernierma** [comment]
Hi,

Things are going in the right direction. A couple of issues were closed this week and some of the main issues are going to be: concerning Automated test (#8), the only question left is how to test the package on windows ; concerning Performance claims (#10), I will send the proofs for my method article this week along with its associated code. The code will be freely available on the JRSSC website so this should settle the Performance claims issue.

**Comment by corybrunson** [comment]
Hi, i apologize for being absent for several days—i had an unreliable connection over the weekend but i should have mentioned that here and affirmed that i'd be back soon. I'm returning to the outstanding issues now.

**Comment by csoneson** [comment]
Hi all - just wanted to check in again. I can see that there are discussions going on in the issues in the source repository - let me know if you have questions where I could help.

**Comment by corybrunson** [comment]
@csoneson i'm sorry for the delay on my part in reviewing the last component i was waiting for, the comparison to other implementations. @fauvernierma has now been able to make this public, so my checklist is complete, but i'd like to wait until @seabbs has commented on that issue to give my recommendation.

**Comment by seabbs** [comment]
I am just waiting on a few small changes to the documentation and need to check @corybrunson thoughts on the testing suite. Once both those are done I will be happy to recommend this for acceptance.

Thanks for checking in @csoneson 

Great job @fauvernierma

**Comment by seabbs** [comment]
@fauvernierma has completed the final changes so my checklist is complete and all review issues are closed. I am now very happy to recommend this for acceptance @csoneson 

Congrats 🎆  @fauvernierma  - this is a really nice package. Looking forward to making use of it in the future.

Hopefully, the review process has been useful!

**Comment by corybrunson** [comment]
@csoneson i also recommend acceptance of the current version. Thank you for your supervision!

**Comment by fauvernierma** [comment]
Thank you very much @csoneson, @seabbs, and @corybrunson  

The review process has been very useful

Have a great day!

**Comment by csoneson** [comment]
@seabbs, @corybrunson - great, thanks a lot!
@fauvernierma - I will have another look through to see if there are any formalities that need to be fixed before I accept the paper.

**Comment by csoneson** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01434/joss.01434/10.21105.joss.01434.pdf)

**Comment by csoneson** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/S0140-6736(17)33326-3 is OK
- 10.1002/sim.4780090506 is OK
- 10.1111/rssc.12368 is OK
- 10.1137/0909024 is OK
- 10.1111/j.1541-0420.2011.01640.x is OK
- 10.1002/sim.2656 is OK
- 10.1017/CBO9780511755453 is OK
- 10.1097/CEJ.0000000000000301 is OK
- 10.1080/01621459.2016.1180986 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by csoneson** [comment]
@fauvernierma - I sent a PR with a couple of small formatting fixes; please check and merge if you agree, then run `@whedon generate pdf` to generate a new version of the paper. 

In addition to that, the next thing to do is to make a tagged release of your software, and create a Zenodo archive. Please make sure that the Zenodo archive has exactly the same title and authors as the paper. Once that's done, please report back here with the DOI of the Zenodo archive.

**Comment by fauvernierma** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01434/joss.01434/10.21105.joss.01434.pdf)

**Comment by fauvernierma** [comment]
Hello, @csoneson 

I would like to make "the CENSUR working survival group" appear as a group author after the name "Laurent Roche", is it possible ?

I thought I did it but obviously it did not work

Thank you

**Comment by csoneson** [comment]
Hi @fauvernierma - sorry, just to clarify, when you say "is it possible", and "it did not work" - do you mean from a technical point of view, or is the question whether it is ok to have a group author?

**Comment by fauvernierma** [comment]
It was a technical question as I do not know how to make it appear on the markdown document

**Comment by csoneson** [comment]
I'm sorry, I'm probably missing the point, but could you not just add "the CENSUR working survival group" as another author in the markdown yaml header?

**Comment by csoneson** [comment]
@fauvernierma - could you please clarify why you would like to add a group author at this stage, and what are the contributions to the submission? According to the JOSS authorship guidelines (https://joss.readthedocs.io/en/latest/submitting.html#authorship), purely financial or organizational contributions, for example, are not sufficient for inclusion among the authors.

**Comment by fauvernierma** [comment]
Of course,

I would like to add the contribution of a working group that all the other authors and I are a part of. This CENSUR project involves our team and other teams from other European countries. It is cited as an author for the method article as it offers guidelines and subjects for future work and the method article and survPen are related to one of said subject.

Do you think it is possible to add the CENSUR project as an author ? I totally understand if it is not

Thank you

**Comment by csoneson** [comment]
Thank you for the clarification! To me this sounds like an "organizational contribution"; however I would like to check with the current AEiC on duty for their opinion as well @openjournals/joss-eics

>I would like to add the contribution of a working group that all the other authors and I are a part of. This CENSUR project involves our team and other teams from other European countries. It is cited as an author for the method article as it offers guidelines and subjects for future work and the method article and survPen are related to one of said subject.
>Do you think it is possible to add the CENSUR project as an author ? I totally understand if it is not

**Comment by arfon** [comment]
> Thank you for the clarification! To me this sounds like an "organizational contribution"; however I would like to check with the current AEiC on duty for their opinion as well @openjournals/joss-eics

I think that's OK. I'm not exactly how our editorial toolchain will handle this but perhaps you can just add an extra author called `CENSUR project team` with an affiliation of `CENSUR project` or something like that?

**Comment by fauvernierma** [comment]
Duely noted

Thank you very much

**Comment by fauvernierma** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01434/joss.01434/10.21105.joss.01434.pdf)

**Comment by fauvernierma** [comment]
It looks good to me but i'll check with the other authors

Thanks again

**Comment by fauvernierma** [comment]
I want to check with them because I am not used to adding an affiliation for this kind of group author

**Comment by csoneson** [comment]
@fauvernierma - just let us know when you and your coauthors are happy with the affiliations, and we'll move on with the final acceptance of your paper!

**Comment by fauvernierma** [comment]
Hi,

It looks almost good to us, I just need to make sure that the CENSUR group won't be considered as the last author.

Typically, if you look at this article in pubmed
https://www.ncbi.nlm.nih.gov/pubmed?term=(Roche%5BAuthor%20-%20Last%5D)%20AND%20%22Statistical%20methods%20in%20medical%20research%22%5BJournal%5D

the CENSUR group appears at the end as a "CORPORATE AUTHOR" but Laurent Roche is considered as the last author.

Can you confirm that for the JOSS article, Laurent Roche will be considered as the last author ?

Thank you very much

**Comment by csoneson** [comment]
I think we need @arfon here, but I don't think that the CENSUR group would be interpreted differently than the other authors. 

> Hi,
> 
> It looks almost good to us, I just need to make sure that the CENSUR group won't be considered as the last author.
> 
> Typically, if you look at this article in pubmed
> https://www.ncbi.nlm.nih.gov/pubmed?term=(Roche%5BAuthor%20-%20Last%5D)%20AND%20%22Statistical%20methods%20in%20medical%20research%22%5BJournal%5D
> 
> the CENSUR group appears at the end as a "CORPORATE AUTHOR" but Laurent Roche is considered as the last author.
> 
> Can you confirm that for the JOSS article, Laurent Roche will be considered as the last author ?
> 
> Thank you very much

**Comment by arfon** [comment]
As currently written, the CENSUR group is the last author. This PR should fix that: https://github.com/fauvernierma/survPen/pull/17

**Comment by fauvernierma** [comment]
Hello again,

I apologize if my request seems weird but this is not really an ordering. The CENSUR group should be treated differently than the "proper" authors.

For example, for the article I cited before, the information concering the article is indexed as follows 

https://www.ncbi.nlm.nih.gov/pubmed?term=(Flexible%5BTitle%5D%20AND%20structured%5BTitle%5D%20AND%20survival%5BTitle%5D%20AND%20model%5BTitle%5D%20AND%20simultaneous%5BTitle%5D%20AND%20estimation%5BTitle%5D%20AND%20non-linear%5BTitle%5D%20AND%20non-proportional%5BTitle%5D%20AND%20effects%5BTitle%5D%20AND%20complex%5BTitle%5D%20AND%20interactions%5BTitle%5D%20AND%20continuous%5BTitle%5D%20AND%20variables%5BTitle%5D%20AND%20Performance%5BTitle%5D%20AND%20multidimensional%5BTitle%5D%20AND%20penalized%5BTitle%5D%20AND%20spline%5BTitle%5D%20AND%20approach%5BTitle%5D%20AND%20net%5BTitle%5D%20AND%20survival%5BTitle%5D%20AND%20trend%5BTitle%5D%20AND%20analysis%5BTitle%5D)&report=medline&format=text

As we can see, CENSUR is not an author (AU) but a collective name (CN). As a matter of fact, it is not even associated with a affiliation

Sorry again for this weird request. If this does not meet the JOSS criteria, maybe it is best to forget about CENSUR.

Thank you

**Comment by arfon** [comment]
Ah OK, thanks for clarifying. I'm afraid this isn't something we can support with our paper metadata. 

It sounds like it might be best to remove CENSUR from the author list.

**Comment by fauvernierma** [comment]
ok it is done (https://github.com/fauvernierma/survPen/commit/0ac67f698ca7a9eb540107a55685e8b557a998e3)

I will create the zenodo archive now

Thank you for your time

**Comment by fauvernierma** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01434/joss.01434/10.21105.joss.01434.pdf)

**Comment by fauvernierma** [comment]
The zenodo archive is available here 
https://zenodo.org/record/3372656#.XVwXIN-6_cs

The corresponding DOI is : 10.5281/zenodo.3372656

Thanks

**Comment by csoneson** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/S0140-6736(17)33326-3 is OK
- 10.1002/sim.4780090506 is OK
- 10.1111/rssc.12368 is OK
- 10.1137/0909024 is OK
- 10.1111/j.1541-0420.2011.01640.x is OK
- 10.1002/sim.2656 is OK
- 10.1017/CBO9780511755453 is OK
- 10.1097/CEJ.0000000000000301 is OK
- 10.1080/01621459.2016.1180986 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by csoneson** [comment]
@whedon set 10.5281/zenodo.3372656 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3372656" target="_blank">10.5281/zenodo.3372656</a> is the archive.

**Comment by csoneson** [comment]
@whedon set v1.2.0 as version

**Comment by whedon** [comment]
OK. v1.2.0 is the version.

**Comment by csoneson** [comment]
@fauvernierma - one final thing. I went through the paper again and found a typo; I fixed it and sent a PR.

**Comment by fauvernierma** [comment]
Ok I just merged it

Thank you @csoneson

**Comment by fauvernierma** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01434/joss.01434/10.21105.joss.01434.pdf)

**Comment by csoneson** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/915

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/915, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by csoneson** [comment]
@openjournals/joss-eics - we are ready to accept this paper!

**Comment by fauvernierma** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @fauvernierma, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by fauvernierma** [comment]
Thank you very much for the whole review process

**Comment by fauvernierma** [comment]
Thank you very much for the whole review process

**Comment by fauvernierma** [comment]
Sorry, I thought it was for me to do it as @csoneson said she was ready to accept the paper

**Comment by fauvernierma** [comment]
Anyway the pdf and XML looks good to me

Thanks again

**Comment by kyleniemeyer** [comment]
The final acceptance is up to the the JOSS editors-in-chief team—I will do some final checks first.

**Comment by fauvernierma** [comment]
Ok, thank you

**Comment by kyleniemeyer** [comment]
@fauvernierma  nearly there! We just need a few changes in the paper:

- One of our paper requirements is:
> Begin your paper with a summary of the high-level functionality of your software for a non-specialist reader. Avoid jargon in this section.

Could you add this at the beginning of your paper, introducing also what survival and net survival analysis are?
- I made two small PRs for the paper and bibliography, if you could merge those: https://github.com/fauvernierma/survPen/pull/19 and https://github.com/fauvernierma/survPen/pull/20

**Comment by fauvernierma** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01434/joss.01434/10.21105.joss.01434.pdf)

**Comment by fauvernierma** [comment]
Hello @kyleniemeyer,

I performed some changes in the article. Let me know if that is what you had in mind

Thank you

**Comment by kyleniemeyer** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/S0140-6736(17)33326-3 is OK
- 10.1002/sim.4780090506 is OK
- 10.1111/rssc.12368 is OK
- 10.1137/0909024 is OK
- 10.1111/j.1541-0420.2011.01640.x is OK
- 10.1002/sim.2656 is OK
- 10.1017/CBO9780511755453 is OK
- 10.1097/CEJ.0000000000000301 is OK
- 10.1080/01621459.2016.1180986 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/923

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/923, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1164933772191682560) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/924
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01434](https://doi.org/10.21105/joss.01434)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats @fauvernierma on your article's publication in JOSS! Many thanks to @csoneson for editing, and @corybrunson and @seabbs for reviewing, this submission.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01434/status.svg)](https://doi.org/10.21105/joss.01434)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01434">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01434/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01434/status.svg
   :target: https://doi.org/10.21105/joss.01434
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01434/status.svg)](https://doi.org/10.21105/joss.01434)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss

**Comment by fauvernierma** [comment]
Hi,

Thanks a lot everyone. I am looking forward to submitting another package to JOSS !

Have a great day !!!!
